Update to newer ioquake3

zturtleman opened the pull request:
(Contains 3 commits from #8 and #9.)

Update ioquake3 to [2023-02-19](https://github.com/ioquake/ioq3/compare/26780805c80c597c3c4209234385979d548d8380...4610a2403a6da423a2d3ac5581a6b6d46298c304) (66 commits)

Project files not changed for Reaction. (The project files are optional; compiling using `make` works on Windows, Linux, and macOS.)

Updates SDL from 2.0.8 to 2.24.0.

zturtleman committed:
Resync to ioquake3 2020-09-09

Reaction was already partically updated to 2020-09-08.
Resync ioquake3 2018-12-21 to 2020-09-09.

zturtleman committed:
Update ioquake3 to 2021-09-27

This is the last ioq3 revision before introducing issues with new
SDL library builds.

zturtleman committed:
Update ioquake3 to 2022-08-18

Updates SDL 2.0.8 to SDL 2.0.16.

SDL build issues; mingw 64-bit is broken, macOS x86 is broken, and
macOS < 10.10 crashes on touchpad scroll. The crash on touchpad scroll
was an SDL bug, fixed in 2.0.18, but the others are issues specific to
ioq3's SDL library builds.

zturtleman committed:
Update ioquake3 to 2023-02-19

Updates SDL 2.0.16 to SDL 2.24.0 and fixes issues with the SDL
library builds.

Mailaender commented:
I can confirm that first commit https://github.com/ReactionQuake3/reaction/pull/11/commits/fe04e3054da840316b2b95a96f4a2dc5a081dca6 resolves https://github.com/ReactionQuake3/reaction/issues/12.

zturtleman committed:
Update ioquake3 to 2024-06-11

zturtleman committed:
Fix emscripten port

Fix USE_OPENAL_DLOPEN=0 and memory for web port and add Reaction file list.

zturtleman commented:
Updated ioquake3 to June 11, 2024 ([ioq3 commits](https://github.com/ioquake/ioq3/compare/4610a2403a6da423a2d3ac5581a6b6d46298c304...c1ab47a7649bed3a9d626833f661fbc6768c1b87)). It fixes a lot of OpenGL2 renderer issues, adds OpenGL ES 2.0+ support to the OpenGL2 renderer, and adds emscripten support for running the game in web browsers (no networking, pk3 caching, or persistent config though).

everystone commented:
Thanks alot @zturtleman , I got the game working again by building this branch.

zturtleman committed:
Update ioquake3 to 2025-07-13 (middle of day)

This merges https://github.com/ioquake/ioq3/compare/c1ab47a7649b...d8b1769de8de
without renaming CHAR_WIDTH to CHARACTER_WIDTH (this was reverted in a
later ioq3 commit).

zturtleman committed:
GitHub CI: Fix building on Windows and macOS

zturtleman commented:
This is likely my last update to this pull request (_Update to newer ioquake3_).

I merged ioquake3 changes up to July 13, 2025 ([ioq3 commits](https://github.com/ioquake/ioq3/compare/c1ab47a7649b...d8b1769de8de)).

This is the last ioquake3 revision that I've merged into my projects. I'm continuing Quake 3 maintenance independently in Lilium Arena. I don't plan to whole-sale merge all ioquake3 changes in the future. So I won't be updating Reaction to newer ioquake3 either.
